tools/interu/src/config/mod.rs

```diff
@@ -0,0 +1,186 @@
+
+        tracing::debug!("deserialize file contents");
+        let config: Config = serde_yaml::from_str(&contents).context(DeserializeSnafu)?;
+
+        config.validate().context(ValidateSnafu)?;
+        Ok(config)
+    }
+
+    #[instrument(name = "validate_config", skip(self))]
+    fn validate(&self) -> Result<(), ValidationError> {
+        for (profile_name, profile) in &self.profiles {
+            tracing::debug!(profile_name, "validate profile");
+
+            profile
+                .validate(profile_name, &self.runners)
+                .context(InvalidProfileConfigSnafu)?;
+        }
+
+        Ok(())
+    }
+
+    pub fn determine_parameters(
+        &self,
+        profile_name: &String,
+        instances: Instances,
+    ) -> Result<Parameters, Error> {
+        // First, lookup the profile by name. Error if the profile does't exist.
+        let profile = self
+            .profiles
+            .get(profile_name)
+            .context(UnknownProfileNameSnafu { profile_name })?;
+
+        // Next, lookup the runner ref based on the profile strategy
+        let runner_ref = match &profile.strategy {
+            profile::Strategy::Weighted(options) => {
+                let weights: Vec<_> = options.weights.iter().map(|w| w.weight).collect();
+                let random_distribution = WeightedIndex::new(weights).unwrap();
+                let mut rng = thread_rng();
+
+                let index = random_distribution.sample(&mut rng);
+                let weight = options.weights.get(index).expect("always valid index");
+
+                &weight.runner
+            }
+            profile::Strategy::UseRunner(options) => &options.runner,
+        };
+
+        // Get the runner based on the runner ref
+        let runner = self.runners.get(runner_ref).unwrap();
+
+        // Get test options
+        let (test_parallelism, test_run, test_parameter) = profile.strategy.get_test_options();
+
+        // Convert our node groups to replicated node groups
+        let node_groups = runner
+            .node_groups
+            .clone()
+            .into_iter()
+            .map(|ng| ReplicatedNodeGroup::try_from(ng, &instances, &runner.platform.distribution))
+            .collect::<Result<Vec<_>, ConvertNodeGroupError>>()
+            .context(ConvertNodeGroupSnafu)?;
+
+        Ok(Parameters {
+            kubernetes_distribution: runner.platform.distribution.clone(),
+            kubernetes_version: runner.platform.version.clone(),
+            test_parameter: test_parameter.to_owned(),
+            test_run: test_run.clone(),
+            test_parallelism,
+            node_groups,
+        })
+    }
+}
+
+#[derive(Debug)]
+pub struct Parameters {
+    kubernetes_distribution: Distribution,
+    kubernetes_version: String,
+
+    node_groups: Vec<ReplicatedNodeGroup>,
+
+    test_parallelism: usize,
+    test_parameter: String,
+    test_run: TestRun,
+}
```

Comment:
Until documented, the fields should have no blank lines around them.

```suggestion
pub struct Parameters {
    kubernetes_distribution: Distribution,
    kubernetes_version: String,
    node_groups: Vec<ReplicatedNodeGroup>,
    test_parallelism: usize,
    test_parameter: String,
    test_run: TestRun,
}
```

Reply:
Most of the Rust code will receive doc comments, including this one.

Reply:
Done on 123a52b